providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/FlagdProviderTest.java

```diff
@@ -57,52 +75,174 @@ public static void init() {
+        final String path = "/some/other/path";
+
+        new EnvironmentVariables("FLAGD_SOCKET_PATH", path).execute(() -> {
+
+            ServiceBlockingStub mockStub = mock(ServiceBlockingStub.class);
+            NettyChannelBuilder mockChannelBuilder = getMockChannelBuilderSocket();
+
+            try (MockedStatic<ServiceGrpc> mockStaticService = mockStatic(ServiceGrpc.class)) {
+                mockStaticService.when(() -> ServiceGrpc.newBlockingStub(any(ManagedChannel.class)))
+                        .thenReturn(mockStub);
+
+                try (MockedStatic<NettyChannelBuilder> mockStaticChannelBuilder = mockStatic(
+                        NettyChannelBuilder.class)) {
+
+                    try (MockedConstruction<EpollEventLoopGroup> mockEpollEventLoopGroup = mockConstruction(
+                            EpollEventLoopGroup.class,
+                            (mock, context) -> {
+                            })) {
+                        mockStaticChannelBuilder.when(() -> NettyChannelBuilder
+                                .forAddress(any(DomainSocketAddress.class))).thenReturn(mockChannelBuilder);
+                        new FlagdProvider();
+
+                        //verify path matches
+                        mockStaticChannelBuilder.verify(() -> NettyChannelBuilder
+                                .forAddress(argThat((DomainSocketAddress d) -> {
+                                    return d.path() == path;
+                                })));
+                    }
+                }
+            }
+        });
+    }
+
+    @Test
+    void host_and_port_arg_should_build_tcp_socket() {
+        final String host = "host.com";
+        final int port = 1234;
+
+        ServiceBlockingStub mockStub = mock(ServiceBlockingStub.class);
+        NettyChannelBuilder mockChannelBuilder = getMockChannelBuilderSocket();
+
+        try (MockedStatic<ServiceGrpc> mockStaticService = mockStatic(ServiceGrpc.class)) {
+            mockStaticService.when(() -> ServiceGrpc.newBlockingStub(any(ManagedChannel.class)))
+                    .thenReturn(mockStub);
+
+            try (MockedStatic<NettyChannelBuilder> mockStaticChannelBuilder = mockStatic(NettyChannelBuilder.class)) {
+
+                mockStaticChannelBuilder.when(() -> NettyChannelBuilder
+                        .forAddress(anyString(), anyInt())).thenReturn(mockChannelBuilder);
+                new FlagdProvider(host, port, false, null);
+
+                // verify host/port matches
+                mockStaticChannelBuilder.verify(() -> NettyChannelBuilder
+                        .forAddress(host, port));
+            }
+        }
+    }
+
+    @Test
+    void no_args_host_and_port_env_set_should_build_tcp_socket() throws Exception {
+        final String host = "server.com";
+        final int port = 4321;
+
+        new EnvironmentVariables("FLAGD_HOST", host, "FLAGD_PORT", String.valueOf(port)).execute(() -> {
```

Comment:
This is a pretty cool library for mocking env vars, might be interesting to you @justinabrahms